ocaml/xapi/create_misc.ml

```diff
@@ -387,7 +387,38 @@ let make_packs_info () =
 		in
 		Array.fold_left (fun l fname -> get_pack_details fname @ l) [] packs
 	with _ -> []
-  
+
+let trim s = String.filter_chars s (not ++ String.isspace)
+
+let make_xen_livepatch_list () = 
+	let lines = String.split '\n' (Helpers.get_process_output "/usr/sbin/xen-livepatch list") in
+ 	let patches = List.fold_left(
+		fun acc l ->
+		match List.map trim (String.split ~limit:2 '|' l) with
+		| [ key; "APPLIED" ] -> key :: acc
+		| _ -> acc;
+	)[] lines in
+	if List.length patches > 0 then Some(String.concat ", " patches) else None
+
+let make_kpatch_list () = 
+	let start_line = "Loaded patch modules:" in 
+	let end_line = "Installed patch modules:" in
+	let lines = String.split '\n' (Helpers.get_process_output "/usr/sbin/kpatch list") in
+	let startpoint = ref false in
+	let patches = List.fold_left(
+		fun acc l ->
+		if !startpoint then begin
+			if String.compare l end_line == 0 then startpoint := false;
+			let trimmed_l = trim l in
+			if !startpoint && String.length trimmed_l > 0 then trimmed_l :: acc else acc
+		end 
+		else begin
+			if String.compare l start_line == 0 then startpoint := true;
+			acc
+		end 
+	)[] lines in
```

Comment:
I tihnk that this would be a little clearer (to OCaml programmers anyway ;)) with a recursion, e.g.:
```
let rec loop acc started = function
  | []                                  -> acc
  | line :: _    when line = end_line   -> acc
  | line :: rest when line = start_line -> loop acc true rest
  | line :: rest ->
    let line' = String.trim line in
    if line' <> "" && started then
      loop (line' :: acc) true rest
    else
      loop acc started rest
in
loop [] false lines
```